Partners showroom and config documentation (#282)
chain4travel/camino-docs: docs/partners/partner-config.md

:::

## Introduction

The Partner Configuration allows Partners to make themselves known to the Camino Network as service providers, or as looking for services from other Partners in the Network. Upon matching between services wanted and offered by two parties, such services will be then traded on Camino Messenger.

Services are defined within the concept of Message Types: you can find [here](../camino-messenger/introduction#main-message-types) a complete documentation with pointers to the list of available types. As a distributor or a supplier, a Partner usually wants _and_ offers several services at the same time.
For example:

A hospitality distributor, e.g. a tour operator, might want to consume:

- AccommodationProductInfoService
- AccommodationProductListService
- AccommodationSearchService
- ValidationService
- MintService

A hospitality supplier, e.g. a hotel chain, might offer the same services for consumption.

> A the same time, the tour operator might want to consume from an info provider:

Reviewer: ```suggestion
At the same time, the tour operator might want to consume from an info provider:
```

## Actors and components

Before dwelwing into the details of how to configure a Partner, it is useful to go through an explanation of involved actors (human and software components, onchain and offchain) and their specific responsibilities in the Partner's operation on Camino Messenger.

```mermaid
graph TD
    CFO([<i>CFO</i>]) --> |owns| FinanceWallet
    CTO([<i>Tech power user</i>]) --> |owns| TechWallet
    CTO --> |deploys| Bot1
    CTO --> |deploys| Bot2
    CTO --> |owns| Bot1Wallet
    CTO --> |owns| Bot2Wallet
    FinanceWallet[<b>Finance wallet</b><br/>- Holds main org funds, allocates them to trading on Messenger] <-.-> CMAccount
    TechWallet[<b>Tech wallet</b><br/>- Creates and manages the CMA, paying gas fees<br/>- Can to withdraw money from CMA] --> CMAccount
    CMAccount[<b>CMA smartcontract</b><br/>- Holds trading funds<br/>- Specifies services and fees<br/>- Specifies accepted currencies<br>- Registers bots] --> Bot1Wallet
    CMAccount --> Bot2Wallet

>     Bot1(<i>Bot1</i>) --> |uses| Bot1Wallet[<b>Bot1 wallet</b><br/>- Can withdraw limited amounts from CMA<br/>- As <i>supplier</i>: cashes in service fee checks, pays gas fees for minting booking tokens<br/>- As <i>distributor</i>: pays for service fees and buying booking tokens]

Reviewer 2: > As distributor: pays for service fees and buying booking tokens

This and also the explanations below about the bots are a little off. Bot's pay only for the gas costs for the cheque cash-in and BookingToken buying operations. The actual amount is paid from the CM Account.

Author: ok, correcting.

    Bot2(<i>Bot2</i>) --> |uses| Bot2Wallet[<b>Bot2 wallet</b><br/>- Same as Bot1 wallet]
```

<br/>The two key roles at a Partner's involved are the Chief Financial Officer (CFO) and a Tech Power User. The CFO is responsible for the Finance Wallet, usually multi-sig, which holds the organization’s main funds. These funds are allocated for trading activities on the Messenger but the CFO never is involved in its operation.

> The Tech power user, on the other hand, is the responsible for the Tech Wallet, which receives funds from the CFO and utilizes them for Messenger operations. By logging in with the Tech Wallet, she is able to create and manage the Partner Configuration, pay for the gas fees involved with setting up and trading on the Camino Messenger, and widhtaw funds, wether to her wallet or to the Finance Wallet.

Reviewer 3: ```suggestion
The Tech power user, on the other hand, is the responsible for the Tech Wallet, which receives funds from the CFO and utilizes them for Messenger operations. By logging in with the Tech Wallet, she is able to create and manage the Partner Configuration, pay for the gas fees involved with setting up and trading on the Camino Messenger, and withdraw funds, whether to her wallet or to the Finance Wallet.
```

The Tech power user (or the Partner's tech departnment) is responsible for the deployments of bots (for an in-depth explanation on bots and how to configure them please refer to [here](../camino-messenger/bot/configuration)). Bots are offchain components that are idenfified and operate on chain through respective wallets, thus having controlled access to the funds in the CMA, with limits on the amounts they can withdraw. The bot wallets facilitate bots' specific blockchain operations. In the case of a supplier's bot, the bot wallet cashes in service fee checks and covers the gas fees for minting booking tokens upon booking confirmation by the partner's systems. In the case of a distributor's bot, the bot wallet handles payments for service fees and the purchase of such booking tokens.

> This flow ensures that the offchain bots can operate within the constraints of the smartcontract, while enabling efficient management of funds and services onchain.

Reviewer 3: here we could mention that we plan a project to have bots auto-fund themselves from the CM account for transaction fees in CAM

Author: this is mentioned in the Manage bots screen, but yes it makes sense to say it also here as people read this first and might wonder.

## Step-by-Step Configuration Guide

This playbook is designed to guide Tech power users through the configuration process for their Camino Messenger Accounts.

1. Register as Partner in the Partner Showroom, or Claim your Partner if it is already listed, as described [here](../partners/introduction).

2. Make sure that your Tech wallet linked to your Partner details:

>    - Is KYC/KYB verified.

Reviewer: KYB is still not supported

Author: Yes but I think we say KYC/KYB everywhere anyway, or please confirm @havan

Reviewer 2: It's ok to have it in the documentation. We have it in a lot of places already.

   - For a listed service, a "Remove Service" button allows you to remove it.

2. **Configure a Service**:
   - **Fee**: You can set a fee to be paid by the caller of this service, displayed in CAM, e.g. 0.05 CAM.
   - **Rack Rates**: A checkbox to indicate whether you're offering "rack" rates (public, non-negotiated rates) for this service.
   - **Capabilities**: A text field to describe the specific capabilities of your service, e.g. "3\* hotels in the Caribbean". An "Add Capability" button allows to add a new capability, while emptying a capability will delete it.
     Note: Setting a Fee and Rack Rates make sense for Search services mostly.

Upon modification, click on "Save Changes" to persist your work on this tab.

### Wanted Services tab

<figure>
<img src="/img/partners/wanted_tab.png" alt="This image displays the screen of the tab where you manage your wanted services."/>
<figcaption align="center">Fig.4: Camino Messenger Account wanted services</figcaption>
</figure>

> The "Wanted Services" screen allows partners to configure and manage the services they ar looking for on the Camino Messenger. Here's a detailed breakdown of the functions that can be enabled on this tab by clicking on "Configure Services":

Reviewer 3: ```suggestion
The "Wanted Services" screen allows partners to configure and manage the services they are looking for on the Camino Messenger. Here's a detailed breakdown of the functions that can be enabled on this tab by clicking on "Configure Services":
```
